### ISSUE TITLE
Error connecting Etherpad-lite instance to Supabase Postgres database

### ISSUE THREAD

**Issue by sbutler-gh** [ether/etherpad-lite#4706]
I'm attempting to use https://supabase.io as a cloud postgres db for an etherpad instance

According to [How to use Etherpad Lite with PostgresSQL](https://github.com/ether/etherpad-lite/wiki/How-to-use-Etherpad-Lite-with-PostgreSQL), this is the correct configuration:

```
"dbType" : "postgres",
"dbSettings" : {
    "user"    : "etherpad",
    "host"    : "localhost",
    "password": "password",
    "database": "etherpad",
    "charset" : "utf8mb4"
}
```

On my https://app.supabase.io/project/[ID]/settings/database page, in the **Connection Info** table, I can view all of those variables

<img width="975" alt="Screen Shot 2021-02-02 at 9 31 23 PM" src="https://user-images.githubusercontent.com/46165064/106689488-5b0b4400-65d0-11eb-8887-9abc169867db.png">

Putting those values in my settings.json file, the configuration looks like this:

```
   "dbType" : "postgres",
   "dbSettings" : {
       "host"    : "db.sxqttvkrwydrmvnlhrtl.supabase.co",
       "database": "postgres",
       "port"    : 5432,
       "user"    : "postgres",
       "password": "[PASSWORD]",
       "charset" : "utf8mb4"
    },
```

When I run **bin/run.sh**, I get the following error:

```
[2021-02-02 21:33:18.443] [INFO] console - Starting Etherpad...
[2021-02-02 21:34:33.916] [ERROR] console - Error: connect ETIMEDOUT 100.26.135.91:5432
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1126:14)
[2021-02-02 21:34:33.917] [ERROR] console - ERROR: Problem while initalizing the database
[2021-02-02 21:34:33.917] [ERROR] console - Error: Connection terminated unexpectedly
    at Connection.<anonymous> (/Users/sam/Projects/etherpad-lite-postgres/src/node_modules/pg/lib/client.js:132:73)
    at Object.onceWrapper (events.js:298:28)
    at Connection.emit (events.js:209:13)
    at Connection.EventEmitter.emit (domain.js:476:20)
    at Socket.<anonymous> (/Users/sam/Projects/etherpad-lite-postgres/src/node_modules/pg/lib/connection.js:58:12)
    at Socket.emit (events.js:209:13)
    at Socket.EventEmitter.emit (domain.js:476:20)
    at TCP.<anonymous> (net.js:658:12)
```

Could it have to do with this configuration setting in supabase, and it's simply taking too long for the connection to be made?

```
authentication_timeout
Sets the maximum allowed time to complete client authentication:              60
```

(In the error log, from `Starting Etherpad` to `Error: connect ETIMEDOut` takes about 75 seconds)

Otherwise, does something else look wrong with my configuration?  Is there a reason that Supabase wouldn't be compatible as a db for Etherpad?  Any other thoughts?

@kiwicopple @awalias Thought you may have some input on the Supabase side

**Comment by rhansen** [comment]
If you run `telnet db.sxqttvkrwydrmvnlhrtl.supabase.co 5432` from your Etherpad server do you get a connection? If not, you have a networking problem such as a misconfigured firewall.

**Comment by kiwicopple** [comment]
>  Is there a reason that Supabase wouldn't be compatible as a db for Etherpad

I can't speak for Etherpad, but our databases are pure Postgres. You can connect using `psql` to test the connection from your machine. If it's not working with `psql` it could be a blockage on your network, or the connection details are incorrect (possibly wrong password). If it's still not working with psql, feel free to send us an email at beta@supabase.io and we will help debug

**Comment by sbutler-gh** [comment]
Hey @rhansen and @kiwicopple , thanks so much for the input.  This definitely put me on the right track.

Per your advice, I wasn't able to get a connection via `telnet` or `psql`, so I figured it's a network issue.

I tried updating my **postgresql.conf** and **pg_hba.conf** files per [this page](https://stackoverflow.com/questions/38466190/cant-connect-to-postgresql-on-port-5432/38466547#38466547), but after doing so, I was still encountering the same issue.

Finally, I realized I had a VPN on.  As soon as I turned off the VPN, I was able to connect via `telnet` and `psql`.  I then reverted my changes to **postgresql.conf** and **pg_hba.conf**, and the connection continued working fine.

So as far as I can understand, the VPN was the issue.  As soon as I turned that off, I was able to connect to my Supabase database and get the Etherpad instance running with `bin/run.sh`, and data is successfully being stored in my Supabase database.

I suppose that means it was just a local network issue with my VPN?  Not sure if there's anything else to glean from the experience -- glad to close the issue out if you all agree
